packages/react-components/src/components/VideoGallery.tsx

```diff
@@ -75,26 +78,51 @@ const DRAG_OPTIONS: IDragOptions = {
   keepInBounds: true
 };
 
+// Sort participants in the following order.
+// 1. Video participants should always render before non-video participants.
+// 2. Video Tiles should be further sorted based on their ordering in dominant speakers list.
 const sortParticipants = (
-  participants: VideoGalleryRemoteParticipant[] | undefined
+  participants?: VideoGalleryRemoteParticipant[],
+  dominantSpeakers?: DominantSpeakers
 ): VideoGalleryRemoteParticipant[] => {
-  if (!participants) {
-    return [];
-  }
+  if (!participants) return [];
 
-  return participants.sort((p1, p2) => {
-    if (!p1?.videoStream?.renderElement?.childElementCount && !p2?.videoStream?.renderElement?.childElementCount) {
-      return 0;
-    }
-    if (!p1?.videoStream?.renderElement?.childElementCount) {
-      return 1;
-    }
-    if (!p2?.videoStream?.renderElement?.childElementCount) {
-      return -1;
-    }
+  const participantsWithVideo: VideoGalleryRemoteParticipant[] = [];
+  const participantsWithoutVideo: VideoGalleryRemoteParticipant[] = [];
 
-    return 0;
+  participants.forEach((p) => {
+    if (p.videoStream?.renderElement?.childElementCount) participantsWithVideo.push(p);
+    else participantsWithoutVideo.push(p);
   });
+
+  // If dominantSpeakers are available, we sort the video tiles basis on dominant speakers.
+  if (dominantSpeakers) {
+    participantsWithVideo.sort((a, b) => {
+      const idxA = dominantSpeakers.speakersList.indexOf(a.userId);
```

Comment:
This potentially has very bad performance because it'll walk `speakersList` once for each `participantWithVideo`.

Instead, convert `speakersList` to a map `userId: index` first.

Reply:
Good Suggestion.